FSJ_django20_project/FSJ/views.py

```diff
@@ -28,13 +28,31 @@ def home(request):
 @user_passes_test(is_FSJ_user)
 def profile(request):
     FSJ_user = get_FSJ_user(request.user.username)
-    if request.POST:
-        profile_form = FSJ_user.get_profile_form(request)
-        FSJ_user.update_FSJUser_from_profile_form(profile_form)
+    if request.method == "POST":
+        if isinstance(FSJ_user, Student):
+            profile_form = StudentRestrictedForm(request.POST, instance=FSJ_user)
+        elif isinstance(FSJ_user, Adjudicator):
+            profile_form = AdjudicatorRestrictedForm(request.POST, instance=FSJ_user)
+        elif isinstance(FSJ_user, Coordinator):
+            profile_form = CoordinatorRestrictedForm(request.POST, instance=FSJ_user)
+        if profile_form.is_valid():
+            profile_form.save()
+            return redirect('profile')
+        else:
+            raise ValueError
+
     else:
-        profile_form = FSJ_user.get_profile_form()
+        if isinstance(FSJ_user, Student):
+            profile_form = StudentRestrictedForm(instance=FSJ_user)
+        elif isinstance(FSJ_user, Adjudicator):
+            profile_form = AdjudicatorRestrictedForm(instance=FSJ_user)
+        elif isinstance(FSJ_user, Coordinator):
+            profile_form = CoordinatorRestrictedForm(instance=FSJ_user)        
+            
     context = get_standard_context(FSJ_user)
     template = loader.get_template("FSJ/profile.html")
     context["form"] = profile_form
+    url = "/profile/"
```

Comment:
Don't need to assign a variable here before adding it to the context instead of adding to context as a string directly.

Reply:
This is for safety reasons, as I've had trouble adding things directly to context before.

Reply:
Ok sounds good. Didn't realize there could be an issue with it, and it does no harm.